### PR TITLE
Add NuMaker-HMI-MA35D1 and RTP porting.

### PR DESCRIPTION
**Pull request by wosayttn**
## 拉取/合并请求描述：(PR description)

(1) Support [MA35D1 heterogeneous multi-core platform](https://github.com/OpenNuvoton/rt-thread/tree/nuvoton/bsp/nuvoton/numaker-hmi-ma35d1)
(2) Regular drivers updating.

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [x] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](../documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/contribution_guide/coding_style_en.txt)

### CODE REVIEW

**Review by mysterywolf — APPROVED** [review]

File: bsp/nuvoton/numaker-hmi-ma35d1/applications/SConscript [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+src = Glob('*.c') + Glob('*.cpp')
+CPPPATH = [cwd, str(Dir('#'))]
+
+group = DefineGroup('Applications', src, depend = [''], CPPPATH = CPPPATH)
+
+objs = objs + group
+
+Return('objs')
+
```
Comment (mysterywolf) [inline bsp/nuvoton/numaker-hmi-ma35d1/applications/SConscript:21]:
```suggestion
from building import *
import os

cwd = GetCurrentDir()
src = Glob('*.c')
CPPPATH = [cwd]

group = DefineGroup('Applications', src, depend = [''], CPPPATH = CPPPATH)

list = os.listdir(cwd)
for item in list:
    if os.path.isfile(os.path.join(cwd, item, 'SConscript')):
        group = group + SConscript(os.path.join(item, 'SConscript'))

Return('group')

```

Comment (mysterywolf) [reply bsp/nuvoton/numaker-hmi-ma35d1/applications/SConscript:21]: 最近统一了一下application文件夹下的sconscript脚本 你看看这个统一的脚本是否合适，如果合适的话就用这个形式就好了：https://github.com/RT-Thread/rt-thread/commit/f7be5fc84b934dfe588acd947fb75abc6833fde8

Comment (wosayttn) [reply bsp/nuvoton/numaker-hmi-ma35d1/applications/SConscript]: Thanks. Synced.

**Comment by BernardXiong** [comment]
Please not use local libcpu. Please use the common libcpu.

**Comment by wosayttn** [comment]
> Please not use local libcpu. Please use the common libcpu.
1. Removed aarch64 libcpu drivers are unused in this board porting.
2. We still keep cortex-a gtimer driver in local. is it possible to revert #4839? The revert also can help #4946. 
3.  For consistent rt-thread device driver porting on both CA35 and CM4, we implement a trapper for CM4.

**Comment by BernardXiong** [comment]
> > Please not use local libcpu. Please use the common libcpu.
> 
> 1. Removed aarch64 libcpu drivers are unused in this board porting.
> 2. We still keep cortex-a gtimer driver in local. is it possible to revert [[libcpu] remove gtimer/pmu from cortex-a #4839](https://github.com/RT-Thread/rt-thread/pull/4839)? The revert also can help [ft2004 bsp 增加gtimer 支持，修正can驱动初始化问题 #4946](https://github.com/RT-Thread/rt-thread/pull/4946).
> 3. For consistent rt-thread device driver porting on both CA35 and CM4, we implement a trapper for CM4.

Em, you can PR the gtimer code. It's better to make more clear and good design for gtimer. About the interrupt trapper, we may loop a Kconfig config for it.

**Comment by wosayttn** [comment]
> > > Please not use local libcpu. Please use the common libcpu.
> > 
> > 
> > 
> > 1. Removed aarch64 libcpu drivers are unused in this board porting.
> > 2. We still keep cortex-a gtimer driver in local. is it possible to revert [[libcpu] remove gtimer/pmu from cortex-a #4839](https://github.com/RT-Thread/rt-thread/pull/4839)? The revert also can help [ft2004 bsp 增加gtimer 支持，修正can驱动初始化问题 #4946](https://github.com/RT-Thread/rt-thread/pull/4946).
> > 3. For consistent rt-thread device driver porting on both CA35 and CM4, we implement a trapper for CM4.
> 
> Em, you can PR the gtimer code. It's better to make more clear and good design for gtimer. About the interrupt trapper, we may loop a Kconfig config for it.

Submitted Revert PR , please approve.
#6491

**Comment by wosayttn** [comment]
Thanks approving.

**Comment by BernardXiong** [comment]
> Thanks approving.

😄 Please keep mind the GTimer new design. Thank you.

**Pull request merged**